### PR TITLE
Feature/tabs addition

### PR DESCRIPTION
**Pull request by necoline**
feature/tabs-addition
Fix for issue #687: Make page anchors on various pages behave like tabs 

Tabs were used to replace Sticky components in country, emergency, region and account views.
react-tabs was brought in as a dependency for main tab features. 
The tab details are stored as an enum to make it easier to map through each tab.
The tab selection is handled in a function in the render content where the tab index is used to select the active tab to display. 

A displayTabs function was added to the componentDidMount to ensure content is displayed in the case of a wrong or missing url param

A tab-content file was created to handle the logic for rendering tab data or a error message in the case of missing data. 

A separate scss file was created for the tab feature and added into the core directory

*** Needs attention:
tab responsiveness
completion of styles to match wire-frame
possible further abstraction of tab component
testing

### CODE REVIEW

File: app/assets/scripts/views/account.js [hunk 1]
```diff
@@ -931,30 +937,40 @@ class Account extends React.Component {
...
-              {this.renderFieldReports()}
-              {this.props.profile.fetched && !this.props.profile.error ? this.renderSubscriptionForm() : null}
-              {this.isPerPermission() ? this.renderPerFormsComponent() : null}
+          <Tabs
+            selectedIndex={TAB_DETAILS.map(({ hash }) => hash).indexOf(this.props.location.hash)}
+            onSelect={index => this.handleTabChange(index)}
+          >
+            <TabList>
+              {TAB_DETAILS.map(tab => (
+                <Tab>{tab.title}</Tab>
```
Comment (danielfdsilva) [inline app/assets/scripts/views/account.js]: `key` is missing

File: app/assets/scripts/components/tab-content.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  errorMessage: 'coming soon',
+  title: 'Error'
+};
+
+if (environment !== 'production') {
+  TabContent.propTypes = {
+    isError: T.boolean,
+    errorMessage: T.string,
+    children: T.object,
+    rest: T.object,
```
Comment (danielfdsilva) [inline app/assets/scripts/components/tab-content.js]: `rest` is not a prop

File: app/assets/scripts/views/account.js [hunk 3]
```diff
@@ -931,30 +937,40 @@ class Account extends React.Component {
...
+                  <TabContent>
+                    {this.renderOperationsFollowing()}
+                    {this.renderAccountInformation()}
+                  </TabContent>
+                </TabPanel>
+                <TabPanel>
+                  <TabContent>
+                    {this.renderFieldReports()}
+                  </TabContent>
+                  <TabContent isError={!this.props.profile.fetched && this.props.profile.error} errorMessage="Subscriptions coming soon" title="Subscriptions">
```
Comment (danielfdsilva) [inline app/assets/scripts/views/account.js]: The content has to be fetched for the error to be available
```suggestion
                  <TabContent isError={this.props.profile.fetched && this.props.profile.error} errorMessage="Subscriptions coming soon" title="Subscriptions">
```

File: app/assets/scripts/views/account.js [hunk 4]
```diff
@@ -931,30 +937,40 @@ class Account extends React.Component {
...
+                    {this.renderSubscriptionForm()}
```
Comment (danielfdsilva) [inline app/assets/scripts/views/account.js]: Should be conditional

```suggestion
                    {this.props.profile.fetched && this.renderSubscriptionForm()}
```

File: app/assets/scripts/views/emergency.js [hunk 5]
```diff
@@ -420,86 +450,107 @@ class Emergency extends React.Component {
...
-            )}
-          </Sticky>
+        {showExportMap()}
+        <Tabs
+          selectedIndex={TAB_DETAILS.map(({ hash }) => hash).indexOf(this.props.location.hash)}
+          onSelect={index => handleTabChange(index)}
+        >
+          <TabList>
+            {TAB_DETAILS.map(tab => (
+              <Tab>{tab.title}</Tab>
```
Comment (danielfdsilva) [inline app/assets/scripts/views/emergency.js]: `key` is missing

File: app/assets/scripts/views/region.js [hunk 6]
```diff
@@ -313,71 +339,92 @@ class AdminArea extends SFPComponent {
...
-              </div>
-            )}
-          </Sticky>
+        <Tabs
+          selectedIndex={TAB_DETAILS.map(({ hash }) => hash).indexOf(this.props.location.hash)}
+          onSelect={index => handleTabChange(index)}
+        >
+          <TabList>
+            {TAB_DETAILS.map(tab => (
+              <Tab>{tab.title}</Tab>
```
Comment (danielfdsilva) [inline app/assets/scripts/views/region.js]: `key` is missing

**Comment by szabozoltan69** [comment]
@danielfdsilva @necoline Is this Pull Request still valid? Or the other one ( https://github.com/IFRCGo/go-frontend/pull/738 ) has done the job? :-)

**Comment by danielfdsilva** [comment]
@szabozoltan69 It would have been better to merge this one, with the cleaner history and all, but since you merged the other one, this is now redundant.

**Comment by szabozoltan69** [comment]
@danielfdsilva we can revert from the other branch (if you thing so), because that https://github.com/IFRCGo/go-frontend/pull/738  was the last one I merged.

I saw 738 mergeable and this 745 not. But if you think this is more clear then let's revert!

**Comment by danielfdsilva** [comment]
@szabozoltan69 Ok, I'll quickly do that

**Comment by szabozoltan69** [comment]
@danielfdsilva Yes, it was a bit too quick. E.g. you have overwritten a former change in account.js: https://github.com/IFRCGo/go-frontend/pull/741/files :-/

**Comment by danielfdsilva** [comment]
@szabozoltan69 That was not because of the revert but because of one of the commits in this pull request that overrode that code. (9ad4aaaffddd44ad683185106f35f78261c31fcf)

As far as I can tell that's the only thing. Created a PR for you: https://github.com/IFRCGo/go-frontend/pull/747

**Comment by szabozoltan69** [comment]
Ok, thanks.

**Pull request merged**